packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts

```diff
@@ -141,13 +287,13 @@ export class GaussianSplattingMesh extends Mesh {
     }
 
     /**
-     * Loads a .splat Gaussian Splatting file asynchronously
+     * Loads a .splat Gaussian or .ply Splatting file asynchronously
      * @param url path to the splat file to load
      * @returns a promise that resolves when the operation is complete
      */
     public loadFileAsync(url: string): Promise<void> {
         return Tools.LoadFileAsync(url, true).then((data) => {
-            this._loadData(data);
+            this._loadData(GaussianSplattingMesh.ConvertPLYToSplat(data));

```

Comment:
Shouldn't you test the extension and call `ConvertPLYToSplat` only if it is ".ply"?

Reply:
this was my 1st draft but nothing guarantees there is a .ply extension anywhere in the url (I believe luma SDK urls don't contain one). Moreover, there is a header check that looks more robust to me.